ocaml/typing/jkind_types.ml

```diff
@@ -28,17 +28,63 @@ module Sort = struct
 
   and var = t option ref
 
-  let equal_const c1 c2 =
-    match c1, c2 with
-    | Void, Void
-    | Value, Value
-    | Float64, Float64
-    | Float32, Float32
-    | Word, Word
-    | Bits32, Bits32
-    | Bits64, Bits64 ->
-      true
-    | (Void | Value | Float64 | Float32 | Word | Bits32 | Bits64), _ -> false
+  module Const = struct
+    type t = const
```

Comment:
I had tried to do this before, but ran into an OCaml issue I couldn't resolve. When making this change, I ended up with `Sort` in `jkind_types.mli` looking like:
```ocaml
module Sort : sig
  module Const : sig
    type t =
      | Void
      | Value
      | Float64
      | Float32
      | Word
      | Bits32
      | Bits64
  end

  type t =
    | Var of var
    | Const of Const.t
  and var = t option ref

  include Jkind_intf.Sort with type t := t and type var := var and type Const.t := Const.t
end
```
This gives an error on the `include` statement:
```
Multiple definition of the module name Const.
Names must be unique in a given structure or signature.
```
Because of this, I believe that we cannot define `Sort.t` in terms of an identifier named `Sort.Const.t`. Or at least, without modifying the `Jkind_intf` and `Jkind_types` structure that we have. Creating a `const` seemed less bad to me.